html/Makefile

```diff
@@ -1,15 +1,53 @@
 SHELL  := bash
+dyn_json := recipes.json archive.json download_counts.json build-status.json
+pages  := $(notdir $(wildcard templates/pages/*.html))
+p_json := $(wildcard packages/*.json)
+p_html := $(p_json:.json=.html)
+htmldir := $(notdir $(CURDIR))
 
-.PHONY: all clean updates.rss
-all: updates.rss
+all: updates.rss pages packages
 
-updates.rss: updates.rss.erb
-	erb updates.rss.erb > updates.rss.tmp && mv updates.rss.tmp updates.rss
+updates.rss: templates/updates.rss.erb packages.json
+	@echo "$(htmldir) • Rendering $@ ..."
+	@./erubi templates/updates.rss.erb > updates.rss.tmp && mv updates.rss.tmp updates.rss
 
-livejson:
-	for f in recipes archive download_counts build-status; do curl https://melpa.org/$$f.json > $$f.json; done
+%.html: templates/pages/%.html templates/layout.html.erb
+	@echo "$(htmldir) • Rendering $@ ..."
+	@./erubi content_template=$< templates/layout.html.erb > $@.tmp && mv $@.tmp $@
+
+index.html: templates/pages/index.html templates/layout.html.erb total_downloads.json packages.json
+	@echo "$(htmldir) • Rendering $@ ..."
+	@./erubi content_template=$< templates/layout.html.erb > $@.tmp && mv $@.tmp $@
+
+packages/%.html: packages/%.json templates/package.html.erb templates/layout.html.erb
+	@echo "$(htmldir) • Rendering $@ ..."
+	@./erubi pname=$* content_template=templates/package.html.erb templates/layout.html.erb > $@.tmp && mv $@.tmp $@
+
+packages/%.json: packages.json
+
+packages.json: archive.json downloads.json generate-packages-json.sh
+	@echo "$(htmldir) • Generating $@ ..."
+	@./generate-packages-json.sh < $< > $@.tmp && mv $@.tmp $@
+
+total_downloads.json: download_counts.json
+	@echo "$(htmldir) • Generating $@ ..."
+	@jq '{ total: add }' $< > $@.tmp && mv $@.tmp $@
+
+downloads.json: download_counts.json ./generate-download-percentile.sh
+	@echo "$(htmldir) • Generating $@ ..."
+	@./generate-download-percentile.sh < $< > $@.tmp && mv $@.tmp $@
+
+pages: $(pages)
+packages: $(p_html)
+
+$(dyn_json):
+	curl -sS https://melpa.org/$@ > $@
```

Comment:
I don't understand this part. This seems like it won't actually do anything except overwrite things? what am I missing?

Reply:
It works the same as the old `livejson` recipe, but without dropping into the shell. Looks like it's for development, which is certainly how I used it. See https://github.com/melpa/melpa/blob/master/html/DEVELOPING.md

Reply:
Oh, my bad. I misread the change.